pkg/frontend/openshiftcluster_delete_test.go

```diff
@@ -0,0 +1,223 @@
+								Properties: api.Properties{
+									ProvisioningState: api.ProvisioningStateDeleting,
+								},
+							},
+						})
+
+						if !m.Matches(doc) {
+							b, _ := json.MarshalIndent(doc, "", "    ")
+							t.Fatal(string(b))
+						}
+
+						return doc, err
+					})
+			},
+			wantStatusCode: http.StatusAccepted,
+			wantAsync:      true,
+		},
+		{
+			name:       "cluster not found in db",
+			resourceID: fmt.Sprintf("/subscriptions/%s/resourcegroups/resourceGroup/providers/Microsoft.RedHatOpenShift/openShiftClusters/resourceName", mockSubID),
+			mocks: func(tt *test, _ *mock_database.MockAsyncOperations, openShiftClusters *mock_database.MockOpenShiftClusters, _ *mock_database.MockSubscriptions) {
+				openShiftClusters.EXPECT().
+					Patch(gomock.Any(), strings.ToLower(tt.resourceID), gomock.Any()).
+					Return(nil, &cosmosdb.Error{StatusCode: http.StatusNotFound})
+			},
+			wantStatusCode: http.StatusNoContent,
+		},
+		{
+			name:       "internal error",
+			resourceID: fmt.Sprintf("/subscriptions/%s/resourcegroups/resourceGroup/providers/Microsoft.RedHatOpenShift/openShiftClusters/resourceName", mockSubID),
+			mocks: func(tt *test, _ *mock_database.MockAsyncOperations, openShiftClusters *mock_database.MockOpenShiftClusters, _ *mock_database.MockSubscriptions) {
+				openShiftClusters.EXPECT().
+					Patch(gomock.Any(), strings.ToLower(tt.resourceID), gomock.Any()).
+					Return(nil, errors.New("random error"))
+			},
+			wantStatusCode: http.StatusInternalServerError,
+			wantError:      `500: InternalServerError: : Internal server error.`,
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			defer cli.CloseIdleConnections()
+
+			l := listener.NewListener()
+			defer l.Close()
+
+			env := env.NewTest(l, clientcerts[0].Raw)
+			env.TLSKey, env.TLSCerts = serverkey, servercerts
+
+			cli.Transport.(*http.Transport).Dial = l.Dial
+
+			controller := gomock.NewController(t)
+			defer controller.Finish()
+
+			asyncOperations := mock_database.NewMockAsyncOperations(controller)
+			openShiftClusters := mock_database.NewMockOpenShiftClusters(controller)
+			subscriptions := mock_database.NewMockSubscriptions(controller)
+
+			tt.mocks(tt, asyncOperations, openShiftClusters, subscriptions)
+
+			f, err := NewFrontend(ctx, logrus.NewEntry(logrus.StandardLogger()), env, &database.Database{
+				AsyncOperations:   asyncOperations,
+				OpenShiftClusters: openShiftClusters,
+				Subscriptions:     subscriptions,
+			}, api.APIs, &noop.Noop{})
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			go f.Run(ctx, nil, nil)
+
+			req, err := http.NewRequest(http.MethodDelete, "https://server"+tt.resourceID+"?api-version=2019-12-31-preview", nil)
+			if err != nil {
+				t.Fatal(err)
+			}
+			resp, err := cli.Do(req)
+			if err != nil {
+				t.Fatal(err)
+			}
+			defer resp.Body.Close()
+
+			if resp.StatusCode != tt.wantStatusCode {
+				t.Error(resp.StatusCode)
+			}
+
+			if tt.wantAsync {
```

Comment:
possibly test that location and azureAsyncOperation are empty strings if !tt.wantAsync?